lib/auth/machineid/machineidv1/bot_service.go

```diff
@@ -686,7 +687,7 @@ func botToUserAndRole(bot *pb.Bot, now time.Time, createdBy string) (types.User,
 	resourceName := BotResourceName(bot.Metadata.Name)
 	role, err := types.NewRole(resourceName, types.RoleSpecV6{
 		Options: types.RoleOptions{
-			MaxSessionTTL: types.Duration(12 * time.Hour),
+			MaxSessionTTL: types.Duration(defaults.MaxRenewableCertTTL),
```

Comment:
Maybe worthy of discussion, a possible alternative to just setting `MaxSessionTTL` to 7 days (which is a bit long) would be to make this an option in the bot proto spec and plumb it through to here. Then users would need to explicitly opt in to the longer max session length on the server for bots to be able to request certs of that length.

I'm already leaning toward that being the better implementation, and if nothing else would probably consider this version as written to be a breaking change that probably should not be backported to v17; I think opt in behavior might be more backport safe.

Reply:
👍  Discussed in slack - sounds like a great plan to me.

Reply:
Thanks for the feedback, I've made the change. We're now keeping old behavior with the automatic 12hr limit, but it can be overridden via the bot resource or via a new flag in `tctl bots add` / `tctl bots update --set-max-session-ttl`.